File: app/channels/messages_channel.rb
class MessagesChannel < ApplicationCable::Channel
  def subscribed
  stream_for Incident.find(params[:id])
  end

  def speak(data)
    incident = Incident.find(params[:id])
    message = current_user.messages.build(content: data["content"], parent_id: data["parentId"], incident: incident)
    MessagesChannel.broadcast_to(incident, message: render(message), parent: message.parent_id) if message.save
  end

  private
    def render(message)
      ApplicationController.renderer.render(partial: "messages/message", locals: { message: message })
    end
end

File: app/javascript/channels/messages_channel.js
import consumer from "./consumer"

$(document).on("turbolinks:load", () => {
    var incident = Number($(".messages").attr("id"));
    if (!isNaN(incident)) {
        consumer.messages = consumer.subscriptions.create({ channel: "MessagesChannel", id: incident }, {
            received(data) {
                if ($("#no-messages-notice")) {
                    $("#no-messages-notice").hide();
                }
                if (data["parent"]) {
                    var replies = $(".message#" + data["parent"]).children(".replies").first();
                    replies.prepend(data["message"]);
                }
                else {
                    $(".messages").prepend(data["message"])
                }
            },
            speak(content, parentId = null) {
                this.perform("speak", { content: content, parentId: parentId })
            }
        });
    }
});
